src/easylink/pipeline_schema_constants.py

```diff
@@ -1,165 +1,123 @@
-                "out_dir": Path("results"),
-            },
-            "in_edges": {
-                "input_data_schema": {
-                    "env_var": "DUMMY_CONTAINER_MAIN_INPUT_FILE_PATHS",
-                    "files": ["file1"],
-                }
-            },
-        },
-        "results_schema": {
-            "step_type": ResultStep,
-            "step_params": {
-                "input_validator": lambda *_: None,
-                "out_dir": Path(),
-            },
-            "in_edges": {
-                "pvs_like_case_study": {
-                    "env_var": "DUMMY_CONTAINER_MAIN_INPUT_FILE_PATHS",
-                    "files": ["result.parquet"],
-                },
-            },
-        },
-    },
-    "development": {
-        "input_data_schema": {
-            "step_type": InputStep,
-            "step_params": {
-                "input_validator": validate_input_file_dummy,
-                "out_dir": Path(),
-            },
-            "in_edges": {},
-        },
-        "step_1": {
-            "step_type": CompositeStep,
-            "step_params": {
-                "step_1": {
-                    "step_type": ImplementedStep,
-                    "step_params": {
-                        "input_validator": validate_input_file_dummy,
-                        "out_dir": Path("intermediate"),
-                    },
-                },
-            },
-            "in_edges": {
-                "input_data_schema": {
-                    "env_var": "DUMMY_CONTAINER_MAIN_INPUT_FILE_PATHS",
-                    "files": ["file1"],
-                }
-            },
-        },
-        "step_2": {
-            "step_type": ImplementedStep,
-            "step_params": {
-                "input_validator": validate_input_file_dummy,
-                "out_dir": Path("intermediate"),
-            },
-            "in_edges": {
-                "step_1": {
-                    "env_var": "DUMMY_CONTAINER_MAIN_INPUT_FILE_PATHS",
-                    "files": ["result.parquet"],
-                }
-            },
-        },
-        "step_3": {
-            "step_type": ImplementedStep,
-            "step_params": {
-                "input_validator": validate_input_file_dummy,
-                "out_dir": Path("intermediate"),
-            },
-            "in_edges": {
-                "step_2": {
-                    "env_var": "DUMMY_CONTAINER_MAIN_INPUT_FILE_PATHS",
-                    "files": ["result.parquet"],
-                }
-            },
+SCHEMA_NODES = [
+    InputStep("input_data_schema", input_slots=[], output_slots=["file1"]),
+    HierarchicalStep(
+        "step_1",
+        input_slots=[
+            (
+                "step_1_main_input",
+                "DUMMY_CONTAINER_MAIN_INPUT_FILE_PATHS",
+                validate_input_file_dummy,
+            )
+        ],
+        output_slots=["step_1_main_output"],
+        nodes=[
+            ImplementedStep(
```

Comment:
nit: I wonder if another name would be clearer for this. I think `BasicStep` was an idea we had.